File: Python/BinaryTrees_BottomViewOfBinaryTree_GFG.py
from collections import deque

'''
Definition for Node
class Node:
    def __init__(self, val):
        self.data = val
        self.right = None
        self.left = None
'''

class Solution:
    def bottomView(self, root):
        answer = []
        
        if root is None:
            return answer
            
        map = {}
        queue = deque()
        
        curr_num = 0
        
        queue.append((root, curr_num))
        
        while queue:
            map[queue[0][1]] = queue[0][0].data
            
            if queue[0][0].left:
                queue.append((queue[0][0].left, queue[0][1] - 1))
            if queue[0][0].right:
                queue.append((queue[0][0].right, queue[0][1] + 1))
                
            queue.popleft()
            
        sorted_map = [map[i] for i in sorted(map.keys())]
        
        for num in sorted_map:
            answer.append(num)
            
        return answer
        

File: C++/BinaryTrees_BottomViewOfBinaryTree_GFG.cpp
/*
Definition for Node
class Node {
public:
    int data;
    Node* left;
    Node* right;

    Node(int val) {
        data = val;
        left = right = nullptr;
    }
};
*/

class Solution {
  public:
    vector<int> bottomView(Node *root) 
    {
        vector<int> answer;
        if(root == NULL)
        {
            return answer;
        }
        
        int curr_num = 0;
        
        map<int, int> myMap;
        queue<pair<Node*, int>> myQueue;
        
        myQueue.push({root, curr_num});
        
        while(!myQueue.empty())
        {
            myMap[myQueue.front().second] = myQueue.front().first -> data;
            
            if(myQueue.front().first -> left)
            {
                myQueue.push({myQueue.front().first -> left, myQueue.front().second - 1});
            }
            if(myQueue.front().first -> right)
            {
                myQueue.push({myQueue.front().first -> right, myQueue.front().second + 1});
            }
            
            myQueue.pop();
        }
        
        for(auto iter : myMap)
        {
            answer.push_back(iter.second);
        }
        
        return answer;
    }
};